docs/source/tutorials/fashion_mnist.ipynb

```diff
@@ -0,0 +1,1274 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "K = 3\n",
+    "\n",
+    "kfold = KFold(n_splits=K, shuffle=True, random_state=SEED)\n",
+    "splits = kfold.split(transformed_dataset)\n",
+    "\n",
+    "train_id_list, test_id_list = [], []\n",
+    "\n",
+    "for fold, (train_ids, test_ids) in enumerate(splits):\n",
+    "    train_id_list.append(train_ids)\n",
+    "    test_id_list.append(test_ids)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Train a model for each fold"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "models = []\n",
+    "\n",
+    "n_epochs = 3\n",
+    "patience = 2\n",
+    "batch_size = 4\n",
+    "\n",
+    "for i in range(K):\n",
+    "    print(f\"Training on fold: {i+1}\")\n",
+    "\n",
+    "    # Create train and test sets and corresponding dataloaders\n",
+    "    trainset = transformed_dataset.select(train_id_list[i])\n",
+    "    testset = transformed_dataset.select(test_id_list[i])\n",
+    "\n",
+    "    trainloader = DataLoader(trainset, batch_size=batch_size, shuffle=True, num_workers=2)\n",
+    "    testloader = DataLoader(testset, batch_size=batch_size, shuffle=False, num_workers=2)\n",
+    "\n",
+    "    model = train(trainloader, testloader, n_epochs, patience)\n",
+    "    models.append(model)\n",
+    "print(\"Finished Training\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 6. Compute out-of-sample predicted probabilities and feature embeddings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pred_probs_list = []\n",
+    "embeddings_list = []\n",
+    "\n",
+    "with torch.no_grad():\n",
+    "    for i in range(K):\n",
+    "        testset = transformed_dataset.select(test_id_list[i])\n",
+    "        testloader = DataLoader(testset, batch_size=batch_size, shuffle=False, num_workers=2)\n",
+    "\n",
+    "        net = models[i].to(device)\n",
+    "        fold_pred_probs, fold_embeddings = [], []\n",
+    "\n",
+    "        for data in tqdm(testloader):\n",
+    "            images, labels = data[\"image\"].to(device), data[\"label\"].to(device)\n",
+    "\n",
+    "            outputs = net(images)\n",
+    "            embeddings = net.embeddings(images)\n",
+    "\n",
+    "            fold_pred_probs.append(outputs.cpu())\n",
+    "            fold_embeddings.append(embeddings.cpu())\n",
+    "\n",
+    "        pred_probs_list.append(torch.vstack(fold_pred_probs))\n",
+    "        embeddings_list.append(torch.vstack(fold_embeddings))\n",
+    "\n",
+    "\n",
+    "# Concatenate features and pred_probs\n",
+    "features = torch.vstack(embeddings_list).numpy()\n",
+    "\n",
+    "logits = torch.vstack(pred_probs_list)\n",
+    "pred_probs = nn.Softmax(dim=1)(logits).numpy()"
```

Comment:
I'd move all this into the earlier loop (training). After model is done training, you can just add the torch-nograd inside the loop and then run this embed and predict code no?